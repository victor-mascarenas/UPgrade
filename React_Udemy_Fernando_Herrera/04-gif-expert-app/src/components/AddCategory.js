import React, {useState} from 'react';

export const AddCategory = () => {
    const [inputValue, setInputValue] = useState('Hola Mundo');

    const inputOnChange = (e) => {
        const newValue = e.target.value;
        setInputValue(newValue);
    };
    const formOnSubmit = (e) => {
        e.preventDefault();
    }

    return (
        <form onSubmit={formOnSubmit}>
            <input type="text" value={inputValue} onChange={inputOnChange}/>
        </form>
    );
};