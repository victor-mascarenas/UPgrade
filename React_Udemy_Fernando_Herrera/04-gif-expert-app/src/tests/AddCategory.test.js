import React from "react";
import {shallow} from 'enzyme';
import AddCategory from '../components/AddCategory';

describe('Pruebas en <AddCategory/>', () => {
    test('Debe de mostrarse correctamete', () => {
        const setCategories = () => {};
        const wrapper = shallow(<AddCategory setCategories={setCategories}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
});
