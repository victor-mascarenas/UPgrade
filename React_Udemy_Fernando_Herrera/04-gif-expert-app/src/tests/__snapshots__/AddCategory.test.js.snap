// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> Debe de mostrarse correctamete 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Add a new category"
    type="text"
    value=""
  />
</form>
`;
